src/components/NewWorkspaceModal.test.js

```diff
@@ -173,4 +199,128 @@ describe('NewWorkspaceModal', () => {
       )
     ).toBeNull();
   });
+
+  it('shows an option for Workspace will have protected data (enhanced bucket logging) if a Google billing project is selected', async () => {
+    // Arrange
+    const user = userEvent.setup();
+
+    Ajax.mockImplementation(() => ({
+      Billing: {
+        listProjects: async () => [gcpBillingProject, azureBillingProject],
+      },
+      ...nonBillingAjax,
+    }));
+
+    await act(async () => {
+      // eslint-disable-line require-await
+      render(
+        h(NewWorkspaceModal, {
+          cloneWorkspace: false,
+          onSuccess: () => null,
+          onDismiss: () => null,
+          customMessage: null,
+          requiredAuthDomain: false,
+          title: null,
+          buttonText: null,
+          // workflowImport: false <== Not specified. False should be the default
+        })
+      );
+    });
+
+    const projectSelector = screen.getByText('Select a billing project');
+    await user.click(projectSelector);
+
+    const googleBillingProject = screen.getByText('Google Billing Project');
+    await user.click(googleBillingProject);
+
+    // Assert
+    // getByText throws an error if the element is not found:
+    screen.getByText('Workspace will have protected data');
```

Comment:
It would be good to add a test that this checkbox is not present if an Azure billing project is selected.